community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java

```diff
@@ -84,11 +84,13 @@ public State init( SessionStateMachine ctx, String clientName, Map<String,Object
                         }
                         catch ( AuthenticationException e )
                         {
-                            return error( ctx, new Neo4jError( e.status(), e.getMessage(), e ) );
+                            ctx.error( new Neo4jError( e.status(), e.getMessage() ));
+                            return halt( ctx );
```

Comment:
I believe making this `return UNINITIALIZED` is sufficient to resolve the concern raised in #7356.